### PR TITLE
documentation for deny statement + privilege levels schema and table

### PR DESCRIPTION
**Pull request by m-kharbat**

### CODE REVIEW

File: blackbox/docs/sql/administration/privileges.txt [hunk 1]
```diff
@@ -12,47 +12,84 @@ Cluster Wide Privileges
...
-User privileges are applied on the whole cluster and propagated
-to all tables and schemas in the cluster. For example, a user with
-``DQL`` privilege can issue ``SELECT`` queries on any table and
-schema in the cluster. Similarly a user with ``DML`` privilege can
-issue ``INSERT`` statement to any table and schema in the cluster.
-There is no option to ``GRANT`` or ``REVOKE`` privileges on object
-level, such as table or schema.
+User privileges can either be applied on a cluster level, they are then
+propagated to all tables and schemas of the cluster, or they can be applied
+on a specific schema or on a specif table.
```
Comment (dobe) [inline blackbox/docs/sql/administration/privileges.txt]: specific schema or table.

File: blackbox/docs/sql/administration/privileges.txt [hunk 2]
```diff
@@ -85,18 +135,67 @@ privileges to a user, namely ``DQL``, ``DML`` and ``DDL``.
...
+To revoke a privilege that was previously granted or denied to a user
+use the :doc:`/sql/reference/revoke` SQL statement, for example::
 
     cr> REVOKE DDL from will;
     REVOKE OK, 1 row affected (... sec)
 
+A privilege can be revoked on a specific schema, like this::
+
+    cr> REVOKE DQL ON SCHEMA sys FROM wolfgang;
+    REVOKE OK, 1 row affected (... sec)
```
Comment (dobe) [inline blackbox/docs/sql/administration/privileges.txt]: document the fact that revoke operates on a certain level in the hierarchy, even the one below which is on cluster. then you also have to document what the rowcount means

File: blackbox/docs/sql/reference/grant.txt [hunk 3]
```diff
@@ -4,25 +4,31 @@
...
+  [{ ON SCHEMA schema_name |
+    ON TABLE  table_ident }]
   TO user_name [, ...];
 
 Description
 ===========
 
 ``GRANT`` is a management statement to grant a ``DQL``, ``DML`` or ``DDL``
-privilege on the cluster to one or many existing users.
+privilege on the cluster, or on a specific schema, or on a specific table
```
Comment (dobe) [inline blackbox/docs/sql/reference/grant.txt]: pls rephrase to more generic e.g: is a management statement to grant a permission on a specific object to users.

File: blackbox/docs/sql/reference/revoke.txt [hunk 4]
```diff
@@ -4,26 +4,31 @@
 ``REVOKE``
 ==========
 
-Revokes privilege from an existing user.
+Revokes a previously granted or denied privilege from an existing user,
+either on the cluster, or on a specific schema, or on a specific table.
```
Comment (dobe) [inline blackbox/docs/sql/reference/revoke.txt]: be more generic

File: blackbox/docs/sql/reference/grant.txt [hunk 5]
```diff
@@ -4,25 +4,29 @@
...
+    ON TABLE  table_ident }]
   TO user_name [, ...];
 
 Description
 ===========
 
-``GRANT`` is a management statement to grant a ``DQL``, ``DML`` or ``DDL``
-privilege on the cluster to one or many existing users.
+``GRANT`` is a management statement to grant one or many privileges
+on a specific object to one or many existing users.
```
Comment (dobe) [inline blackbox/docs/sql/reference/grant.txt]: the docs of grant, deny and revoke docs do not cover the cluster case

File: blackbox/docs/sql/reference/deny.txt [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+Synopsis
+========
+
+.. code-block:: psql
+
+  DENY { { DQL | DML | DDL [,...] } | ALL [ PRIVILEGES ] }
+  [{ ON SCHEMA schema_name [, ...] |
+    ON TABLE  table_ident [, ...]  }]
+  TO user_name [, ...];
+
```
Comment (dobe) [inline blackbox/docs/sql/reference/deny.txt]: i think this the synopsis needs to be changed in that that schema_name and table_ident are actually just identifiers. e.g:

... ON {TABLE | SCHEMA} identifier ....

additionally the cluster case needs to be documented here.

the same applies to REVOKE and GRANT

File: blackbox/docs/sql/administration/privileges.txt [hunk 7]
```diff
@@ -12,47 +12,84 @@ Cluster Wide Privileges
...
+    GRANT OK, 1 row affected (... sec)
+
+This statement will deny ``DQL`` privilege to user riley on the ``doc`` schema table
+``doc.accounting``. However, user riley, will still have ``DQL`` privilege on all the
+other tables of the ``doc`` schema.
+
+.. note::
+
+    Privileges can be managed on all schemas and tables of the cluster,
+    including the ``sys`` schema.
```
Comment (mfussenegger) [inline blackbox/docs/sql/administration/privileges.txt]: Isn't the `sys.users` table still kind of special in that it's super-user only? Or will it be possible to grant users permission to it as well?

Comment (m-kharbat) [inline blackbox/docs/sql/administration/privileges.txt]: In the story, it will also be possible to configure, the sys schema and the sys tables privileges

**Pull request merged**